src/Core/Project.cs

```diff
@@ -19,30 +19,40 @@
 #endregion
 
 using Reko.Core.Lib;
+using Reko.Core.Loading;
 using Reko.Core.Scripts;
-using System;
-using System.Collections.Generic;
-using System.Collections.Specialized;
-using System.Collections.ObjectModel;
-using System.IO;
-using System.Linq;
-using System.Text;
 
 namespace Reko.Core
 {
     /// <summary>
     /// A Reko decompiler project.
     /// </summary>
-    public class Project 
+    public class Project : ILoadedImage
     {
         public Project()
         {
             Programs = new ObservableRangeCollection<Program>();
             MetadataFiles = new ObservableRangeCollection<MetadataFile>();
             ScriptFiles = new ConcurrentObservableCollection<ScriptFile>();
             LoadedMetadata = new TypeLibrary();
+            Location = default!;
```

Comment:
We are lying here. And it caused error at https://github.com/uxmal/reko/pull/1098#discussion_r761205395. Maybe it would be better to add explicit constructor parameter in the same way as for `AbstractHierarchicalArchive`

Reply:
"What is the cost of lies?"

I've added a constructor to `Project` that accepts an `ImageLocation`. All production code will use that. The parameter-less constructor can be used by unit tests that do don't use `ILoader`.